server/autolink/lib_jira_test.go

```diff
@@ -80,43 +80,75 @@ var jiraTests = []linkTest{
 	}, {
 		"Url replacement",
 		autolink.Autolink{
-			Pattern:  "(https://mattermost.atlassian.net/browse/)(MM)(-)(?P<jira_id>\\d+)",
-			Template: "[MM-$jira_id](https://mattermost.atlassian.net/browse/MM-$jira_id)",
+			Pattern:  "(https://mattermost.atlassian.net/browse/)(?P<project_id>\\w+)(-)(?P<jira_id>\\d+)",
+			Template: "[${project_id}-${jira_id}](https://mattermost.atlassian.net/browse/${project_id}-${jira_id})",
 		},
 		"Welcome https://mattermost.atlassian.net/browse/MM-12345 should link!",
 		"Welcome [MM-12345](https://mattermost.atlassian.net/browse/MM-12345) should link!",
 	}, {
 		"Url replacement multiple times",
 		autolink.Autolink{
-			Pattern:  "(https://mattermost.atlassian.net/browse/)(MM)(-)(?P<jira_id>\\d+)",
-			Template: "[MM-$jira_id](https://mattermost.atlassian.net/browse/MM-$jira_id)",
+			Pattern:  "(https://mattermost.atlassian.net/browse/)(?P<project_id>\\w+)(-)(?P<jira_id>\\d+)",
+			Template: "[${project_id}-${jira_id}](https://mattermost.atlassian.net/browse/${project_id}-${jira_id})",
 		},
 		"Welcome https://mattermost.atlassian.net/browse/MM-12345. should link https://mattermost.atlassian.net/browse/MM-12346 !",
 		"Welcome [MM-12345](https://mattermost.atlassian.net/browse/MM-12345). should link [MM-12346](https://mattermost.atlassian.net/browse/MM-12346) !",
 	}, {
 		"Url replacement multiple times and at beginning",
 		autolink.Autolink{
-			Pattern:  "(https:\\/\\/mattermost.atlassian.net\\/browse\\/)(MM)(-)(?P<jira_id>\\d+)",
-			Template: "[MM-$jira_id](https://mattermost.atlassian.net/browse/MM-$jira_id)",
+			Pattern:  "(https:\\/\\/mattermost.atlassian.net\\/browse\\/)(?P<project_id>\\w+)(-)(?P<jira_id>\\d+)",
+			Template: "[${project_id}-${jira_id}](https://mattermost.atlassian.net/browse/${project_id}-${jira_id})",
 		},
 		"https://mattermost.atlassian.net/browse/MM-12345 https://mattermost.atlassian.net/browse/MM-12345",
 		"[MM-12345](https://mattermost.atlassian.net/browse/MM-12345) [MM-12345](https://mattermost.atlassian.net/browse/MM-12345)",
 	}, {
 		"Url replacement at end",
 		autolink.Autolink{
-			Pattern:  "(https://mattermost.atlassian.net/browse/)(MM)(-)(?P<jira_id>\\d+)",
-			Template: "[MM-$jira_id](https://mattermost.atlassian.net/browse/MM-$jira_id)",
+			Pattern:  "(https://mattermost.atlassian.net/browse/)(?P<project_id>\\w+)(-)(?P<jira_id>\\d+)",
+			Template: "[${project_id}-${jira_id}](https://mattermost.atlassian.net/browse/${project_id}-${jira_id})",
 		},
 		"Welcome https://mattermost.atlassian.net/browse/MM-12345",
 		"Welcome [MM-12345](https://mattermost.atlassian.net/browse/MM-12345)",
+	}, {
+		"Comment url replacement",
+		autolink.Autolink{
+			Pattern:  "(https://mattermost.atlassian.net/browse/)(?P<project_id>\\w+)(-)(?P<jira_id>\\d+)[?](focusedCommentId)(=)(?P<comment_id>\\d+)",
+			Template: "[${project_id}-${jira_id} With Comment #${comment_id}](https://mattermost.atlassian.net/browse/${project_id}-${jira_id}?focusedCommentId=${comment_id})",
```

Comment:
Not sure if having the comment id in the link text is that helpful. The capitalization of "With Comment" seems weird too.

Reply:
@mickmister can you suggest a message for this , planning to raise a PR to fix your comments.

Reply:
How about:

```
[${project_id}-${jira_id} (comment)](https://mattermost.atlassian.net/browse/${project_id}-${jira_id}?focusedCommentId=${comment_id})
```

which would render as

[MM-1337 (comment)](https://mattermost.atlassian.net/browse/${project_id}-${jira_id}?focusedCommentId=${comment_id})

This is how GitHub does their comment links, such as https://github.com/mattermost/mattermost-plugin-autolink/pull/185#discussion_r841081248